capa/features/extractors/dnfile/insn.py

```diff
@@ -38,55 +41,59 @@ def get_imports(ctx: Dict) -> Dict:
     return ctx["imports_cache"]
 
 
-def extract_insn_api_features(f: CilMethodBody, bb: CilMethodBody, insn: Instruction) -> Iterator[Tuple[API, int]]:
+def extract_insn_api_features(fh: FunctionHandle, bh, ih: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
     """parse instruction API features"""
+    f: CilMethodBody = fh.inner
+    insn: Instruction = ih.inner
+
     if insn.opcode not in (OpCodes.Call, OpCodes.Callvirt, OpCodes.Jmp, OpCodes.Calli):
         return
 
-    name: str = get_imports(f.ctx).get(insn.operand.value, "")
+    name: str = get_imports(fh.ctx).get(insn.operand.value, "")
     if not name:
         return
 
     if "::" in name:
         # like System.IO.File::OpenRead
-        yield API(name), insn.offset
+        yield API(name), ih.address
     else:
         # like kernel32.CreateFileA
         dll, _, symbol = name.rpartition(".")
         for name_variant in capa.features.extractors.helpers.generate_symbols(dll, symbol):
-            yield API(name_variant), insn.offset
+            yield API(name_variant), ih.address
 
 
-def extract_insn_number_features(
-    f: CilMethodBody, bb: CilMethodBody, insn: Instruction
-) -> Iterator[Tuple[Number, int]]:
+def extract_insn_number_features(fh, bh, ih: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
     """parse instruction number features"""
+    insn: Instruction = ih.inner
+
     if insn.is_ldc():
-        yield Number(insn.get_ldc()), insn.offset
+        yield Number(insn.get_ldc()), ih.address
 
 
-def extract_insn_string_features(
-    f: CilMethodBody, bb: CilMethodBody, insn: Instruction
-) -> Iterator[Tuple[String, int]]:
+def extract_insn_string_features(fh: FunctionHandle, bh, ih: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
     """parse instruction string features"""
+    f: CilMethodBody = fh.inner
+    insn: Instruction = ih.inner
+
     if not insn.is_ldstr():
         return
 
     if not isinstance(insn.operand, StringToken):
         return
 
-    user_string: Optional[str] = read_dotnet_user_string(f.ctx["pe"], insn.operand)
+    user_string: Optional[str] = read_dotnet_user_string(fh.ctx["pe"], insn.operand)
     if user_string is None:
         return
 
-    yield String(user_string), insn.offset
+    yield String(user_string), ih.address
 
 
-def extract_features(f: CilMethodBody, bb: CilMethodBody, insn: Instruction) -> Iterator[Tuple[Feature, int]]:
+def extract_features(f: FunctionHandle, bb: BBHandle, insn: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
```

Comment:
need to look out for consistent names (`f`, `fh`, `fhandle`, etc.) although the type annotations help